### PR TITLE
update the provisional extension notification text

### LINKED ISSUE #971: update provisional extension text
We currently include the following text in the "version history" for each of our provisional extensions (example [link](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_Ext.html#_version_history_42)):

>This is a preview of an OpenCL provisional extension specification that has been Ratified under the Khronos Intellectual Property Framework. It is being made publicly available prior to being uploaded to the Khronos registry to enable review and feedback from the community. If you have feedback please create an issue on https://github.com/KhronosGroup/OpenCL-Docs/

This text is helpful but it could stll be improved.  Some examples:

1. We should say explicitly that provisional extensions can change in non-backward compatible ways.  Related to the discussion on https://github.com/KhronosGroup/OpenCL-Headers/pull/238.
2. This text is in the extension specification, which is hosted on the Khronos registry, so there should be a minor update to the "prior to be uploaded to the Khronos registry" text.
3. Any other updates and improvements - it's been a while since we first drafted this text!

### PR DESCRIPTION
**Pull request by bashbaug**
fixes #971 

* Factors the provisional extension notification text into a separate file to avoid copy-paste.
* Updates the provisional extension text to reflect that provisional extensions are published on the OpenCL registry but may change in non-backward compatible ways.
* Includes the provisional extension text consistently in all provisional extensions, below the version table information.

### CODE REVIEW

**Comment by bashbaug** [comment]
@kpet here's one more where it'd be great to have your approval before merging - thanks!

**Review by kpet — APPROVED** [review]
Nice step forward!

**Pull request merged**
